pkg/platform/platform.go

```diff
@@ -35,15 +35,15 @@ func InfraStatus(client cnoclient.Client) (*bootstrap.InfraStatus, error) {
 		APIServers:             map[string]bootstrap.APIServer{},
 	}
 
-	// If we use hypershift, the components need proxy settings. In selfhosted, they
-	// are always local to the API so there it is never needed.
-	if res.ControlPlaneTopology == configv1.ExternalTopologyMode {
-		proxy := &configv1.Proxy{}
-		if err := client.Default().CRClient().Get(context.TODO(), types.NamespacedName{Name: "cluster"}, proxy); err != nil {
-			return nil, fmt.Errorf("failed to get proxy 'cluster': %w", err)
-		}
-		res.Proxy = proxy.Status
+	// If we use hypershift, the components need proxy settings. In selfhosted,
+	// only the CNCC needs proxy setttings.
+	proxy := &configv1.Proxy{}
+	if err := client.Default().CRClient().Get(
+		context.TODO(), types.NamespacedName{Name: "cluster"}, proxy,
+	); err != nil && res.ControlPlaneTopology == configv1.ExternalTopologyMode {
+		return nil, fmt.Errorf("failed to get proxy 'cluster': %v", err)
```

Comment:
So I wonder if this is even correct for hypershift - shall we really fail if the proxy/cluster object cannot be found? Shouldn't we merely log a warning?

Reply:
We deny cluster configs deletion: https://github.com/openshift/kubernetes/blob/master/openshift-kube-apiserver/admission/customresourcevalidation/config/deny_delete_cluster_config_resource.go#L31

If there is no custom proxy configuration `Proxy` has an empty spec:
```
apiVersion: config.openshift.io/v1
kind: Proxy
metadata:
  creationTimestamp: "2022-10-06T07:26:01Z"
  generation: 1
  name: cluster
  resourceVersion: "450"
  uid: 07de649c-391d-4937-9af7-e5582f8324b4
spec:
  trustedCA:
    name: ""
status: {}
```

I think we should treat `Proxy` the same as `Infrastructure` and always return an error if get failed.